[FIX] web: fix the name is misaligned

shup-odoo opened the pull request:
Steps to reproduce:
----------------------------------------------------------------------------------------
1. Install project.
2. Go to the project.
3. Create or edit the project name too long.
4. Then go to Configuration and select Task stages.
5. The project name is slightly misaligned in list view.

Issue:
----------------------------------------------------------------------------------------
- Before this commit, in the project's task stages, if the project's name
 was long, it would misalign in list view.

Cause:
----------------------------------------------------------------------------------------
- When the name is too long it will be laid out in a row.

Fix:
----------------------------------------------------------------------------------------
- After this commit the name is now aligned by applying the 'd-inline-flex'
 Bootstrap class.

task-3681318

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/151705).

pkyriakou reviewed (commented):
Thank you @shup-odoo LGTM :100:

xavierbol commented on addons/web/static/src/views/list/list_renderer.xml:
```diff
@@ -171,9 +171,9 @@
+                    <span t-attf-class="o_group_caret fa {{group.isFolded ? 'fa-caret-right' : 'fa-caret-down' }}"
                         t-attf-style="--o-list-group-level: {{getGroupLevel(group)}}"/>
-                    <t t-esc="getGroupDisplayName(group)"/> (<t t-esc="group.count"/>)
+                        <span class="text-truncate ms-2 me-1" t-esc="getGroupDisplayName(group)"/> (<t t-esc="group.count"/>)
```
The tabulation added is not needed, IMO.

Also, I am not sure we could do that in stable, let's see what Framework JS thinks.

robodoo commented:
@shup-odoo the target branch 'saas-16.3' has been disabled, you may want to close this PR.

xavierbol commented:
Hello @shup-odoo 

Can you check if we still have the issue in 17+? 🧐

shup-odoo commented:
Hello @xavierbol 
There is still an issue in versions 17 through master.
Thank you !!!

xavierbol commented:
Can you retarget your fix then? 

Thanks in advance!

shup-odoo commented:
Hello @xavierbol 

I've made changes please have a look. Enterprise one test is failing due to our fix so I've changed it and added the PR in task description. Let me know if I've missed anything.

Thanks in advance :smile:

xavierbol commented on addons/web/static/src/views/list/list_renderer.xml:
```diff
@@ -176,9 +176,9 @@
+                    <span t-attf-class="o_group_caret fa {{group.isFolded ? 'fa-caret-right' : 'fa-caret-down' }}"
                         t-attf-style="--o-list-group-level: {{getGroupLevel(group)}}"/>
-                    <t t-esc="getGroupDisplayName(group)"/> (<t t-esc="group.count"/>)
+                    <span class="text-truncate ms-2 me-1" t-esc="getGroupDisplayName(group)"/> (<t t-esc="group.count"/>)
```
Why do we need to change the XML tag? It does not really respect the stable policy. 😕 

Cannot we fix the issue without changing the XML tag?

ppr-odoo replied:
Yes, the stable policy is being violated. Currently, class is not applied to the `<t>` tag. can we make the invisible?

``<t class="o_data" t-esc="getGroupDisplayName(group)" t-if="1 == 0"/>``

ppr-odoo commented:
@xavierbol This visual improvement. Can we do it in the master?

xavierbol commented:
Then it should be better to improve in master. 

It is not really a big issue, so it is acceptable to only do it in master.
